server/src/main/java/org/opensearch/index/snapshots/blobstore/RemoteStoreShardShallowCopySnapshot.java

```diff
@@ -0,0 +1,326 @@
+                    primaryTerm = parser.longValue();
+                } else if (PARSE_COMMIT_GENERATION.match(currentFieldName, parser.getDeprecationHandler())) {
+                    commitGeneration = parser.longValue();
+                } else if (PARSE_START_TIME.match(currentFieldName, parser.getDeprecationHandler())) {
+                    startTime = parser.longValue();
+                } else if (PARSE_TIME.match(currentFieldName, parser.getDeprecationHandler())) {
+                    time = parser.longValue();
+                } else if (PARSE_TOTAL_FILE_COUNT.match(currentFieldName, parser.getDeprecationHandler())) {
+                    totalFileCount = parser.intValue();
+                } else if (PARSE_TOTAL_SIZE.match(currentFieldName, parser.getDeprecationHandler())) {
+                    totalSize = parser.longValue();
+                } else if (PARSE_INDEX_UUID.match(currentFieldName, parser.getDeprecationHandler())) {
+                    indexUUID = parser.text();
+                } else if (PARSE_REMOTE_STORE_REPOSITORY.match(currentFieldName, parser.getDeprecationHandler())) {
+                    remoteStoreRepository = parser.text();
+                } else if (PARSE_REPOSITORY_BASE_PATH.match(currentFieldName, parser.getDeprecationHandler())) {
+                    repositoryBasePath = parser.text();
+                } else {
+                    throw new OpenSearchParseException("unknown parameter [{}]", currentFieldName);
+                }
+            } else if (token == XContentParser.Token.START_ARRAY) {
+                if (PARSE_FILE_NAMES.match(currentFieldName, parser.getDeprecationHandler())) {
+                    while ((parser.nextToken()) != XContentParser.Token.END_ARRAY) {
+                        fileNames.add(parser.text());
+                    }
+                } else {
+                    parser.skipChildren();
+                }
+            } else {
+                parser.skipChildren();
+            }
+        }
+
+        // Verify that file information is complete
+        if (snapshot == null) {
+            throw new OpenSearchParseException("missing Snapshot Name");
+        }
+        if (indexVersion < 0) {
+            throw new OpenSearchParseException("missing or invalid Index Version");
+        }
+        if (commitGeneration < 0) {
+            throw new OpenSearchParseException("missing or invalid Commit Generation");
+        }
+        if (primaryTerm < 0) {
+            throw new OpenSearchParseException("missing or invalid Primary Term");
+        }
+        if (indexUUID == null) {
+            throw new OpenSearchParseException("missing Index UUID");
+        }
+        if (remoteStoreRepository == null) {
+            throw new OpenSearchParseException("missing Remote Store Repository");
+        }
+        if (repositoryBasePath == null) {
+            throw new OpenSearchParseException("missing Repository Base Path");
+        }
```

Comment:
You can move these mandatory parameter checks in constructor or separate method like `verifyParameters` which is called from constructor. For `null` validations we can use `Objects.requireNonNull()`